presto-orc/src/main/java/com/facebook/presto/orc/reader/BatchStreamReaders.java

```diff
@@ -40,7 +40,7 @@ public static BatchStreamReader createStreamReader(Type type, StreamDescriptor s
             case INT:
             case LONG:
             case DATE:
-                return new LongBatchStreamReader(type, streamDescriptor, systemMemoryContext);
+                return new LongBatchStreamReader(type, streamDescriptor, systemMemoryContext, options.isResetAllReaders());
```

Comment:
Why are you changing batch reader? They are not used in Spark.

Reply:
Right, those are not used in spark. but I think not reset all underlying readers is a generic bug, hence including them. Let me know if you prefer to keep the current behavior in this PR.

Reply:
Batch reader is rarely used in prod. I'd suggest to not touch it.